sashimi-webapp/src/logic/documentPackager/conditionalProcessor.js

```diff
@@ -0,0 +1,240 @@
+      ) {
+    return false;
+  }
+
+  // Check if got category e.g. ==>hide:slides ...
+  if (state.src.charCodeAt(pos + 7) === 58  /* : */) {
+    hasCategory = true;
+    category = state.src.slice(pos+8, max).toString();
+  }
+
+  mem = pos;
+  pos = state.skipChars(pos, state.src.charCodeAt(pos + 6));
+
+  len = pos - mem;
+
+  markup = state.src.slice(mem, pos);
+  params = state.src.slice(pos, max);
+
+  // Since start is found, we can report success here in validation mode
+  if (silent) { return true; }
+
+  // search end of block
+  nextLine = startLine;
+
+  if (!hasCategory || category === fileName) {
+    for (;;) {
+      nextLine += 1;
+      if (nextLine >= endLine) {
+        // unclosed block should be leave the ==>hide there
+        return false;
+      }
+
+      pos = state.bMarks[nextLine] + state.tShift[nextLine];
+      max = state.eMarks[nextLine];
+
+      // This checks if there was an unclosed hide block
+      // before the start of another hide block
+      // e.g. ==>hide *Text here* \n\n ==>hide ... <==
+      if (state.src.charCodeAt(pos) === 61  /* = */&&
+      state.src.charCodeAt(pos + 1) === 61  /* = */&&
+      state.src.charCodeAt(pos + 2) === 62 /* > */ &&
+      state.src.charCodeAt(pos + 3) === 104 /* h */ &&
+      state.src.charCodeAt(pos + 4) === 105 /* i */ &&
+      state.src.charCodeAt(pos + 5) === 100 /* d */ &&
+      state.src.charCodeAt(pos + 6) === 101 /* e */
+      ) {
+        return false;
+      }
+
+      if (pos < max && state.sCount[nextLine] < state.blkIndent) {
+          // non-empty line with negative indent should stop the list:
+          // - <==
+          //  test
+        break;
+      }
+
+      marker = state.src.charCodeAt(pos);
+      const marker2 = state.src.charCodeAt(pos + 1);
+      const marker3 = state.src.charCodeAt(pos + 2);
+
+      if (marker === 0x3C /* < */ && marker2 === 0x3D /* = */ &&
+          marker3 === 0x3D /* = */
+      ) {
+        pos = state.skipChars(pos, marker3);
+
+        // make sure tail has spaces only
+        pos = state.skipSpaces(pos);
+
+        if (pos >= max) { break; }
+
+        // found!
+        haveEndMarker = true;
+        // state.md.block.tokenize(state, startLine + 1, nextLine - 1);
+        break;
+      }
+    }
+  } else if (category !== fileName) {
+    // Having found the start hide syntax (e.g. ==>hide:notequal), find the
+    // end marker (e.g. <==) and remove them from output into viewer, since
+    // the content will not be hidden
+    const contentStartLine = nextLine + 1;
+    let oldLineMax;
+
+    for (;;) {
+      nextLine += 1;
+
+      pos = state.bMarks[nextLine] + state.tShift[nextLine];
+      max = state.eMarks[nextLine];
+
+      marker = state.src.charCodeAt(pos);
+      const marker2 = state.src.charCodeAt(pos + 1);
+      const marker3 = state.src.charCodeAt(pos + 2);
+
+      if (marker === 0x3C /* < */ && marker2 === 0x3D /* = */ &&
+          marker3 === 0x3D /* = */
+      ) {
```

Comment:
It is possible for this code to stuck in an infinity loop.

case: When user are in progress of typing out the condition without a `<==` tag at the back
```markdown
==>hide:
```